app/api-fetch/all-songs.ts

```
@@ -2,26 +2,43 @@ import axios from "axios";
 
 export default async function fetchAllSongs() {
     try {
-        const res = await axios.get(`${process.env.NEXT_PUBLIC_API_URL}/v1/song?page=1&limit=10`);
-        const data = SongSchema.parse(res.data);
-        return data["data"] as Song[];
+        localStorage.getItem("songs");
+        if (localStorage.getItem("songs") !== null && localStorage.getItem("songsTime") === null) {
+            localStorage.removeItem("songs");
+        }
+        // xoá cache nếu đã quá 1 phút
+        if (localStorage.getItem("songs") !== null || Date.now() - parseInt(localStorage.getItem("songsTime")!) < 3600000) {
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)